scripts/TRN(registry data).R

```
@@ -0,0 +1,415 @@
+  group_by(id) |>
+  summarize(trns_reg = paste(crossreg_trn, collapse = ";")) |>
+  mutate(protocol_sponsor_code = NA) |>
+  mutate(results_sponsor_code = NA) |>
+  relocate(protocol_sponsor_code, .before = trns_reg) |>
+  relocate(results_sponsor_code, .before = trns_reg) |>
+  mutate(is_primary_IV_id = TRUE) |>
+  mutate(who_utn = NA)
+
+
+##########################################################
+
+# Clean and prepare EU protocol trns separately before merging with IV_clean
+
+EU_protocol_clean <- EU_protocol_dump |>
+                     select(eudract_number,
+                            sponsor_s_protocol_code_number,
+                            isrctn_international_standard_randomised_controlled_trial_numbe,
+                            us_nct_clinicaltrials_gov_registry_number,
+                            who_universal_trial_reference_number_utrn,
+                            other_identifiers) |>
+                     rename(isrctn_number =  isrctn_international_standard_randomised_controlled_trial_numbe,
+                            nct_number = us_nct_clinicaltrials_gov_registry_number,
+                            who_utn_number = who_universal_trial_reference_number_utrn,
+                            other_ids = other_identifiers,
+                            protocol_sponsor_code = sponsor_s_protocol_code_number)
+
+# Columns we want cleaned by Maia's script
+# got rid of WHO number column (who_universal_trial_reference_number_utrn) as Maia said to keep it separate, plus the well formatted numbers here still return NA with which_registry()
+# which will mess up our logic
+
+protocol_columns_to_clean <- c("isrctn_number",
+                               "nct_number",
+                               "other_ids")
+
+# Initialize separate unclean_ids columns for each column to be cleaned
+for (col in protocol_columns_to_clean) {
+  unclean_col <- paste0(col, "_protocol_unclean")
+  EU_protocol_clean[[unclean_col]] <- NA_character_
```

in some cases I'm seeing NA as character and other times as logical. I believe this should work since the original data type in `EU_dump` is chr(), but check if any implications or silent errors due to this.